packages/superset-ui-dimension/test/getMultipleTextDimensions.test.ts

```diff
@@ -191,15 +191,13 @@ describe('getTextDimension(input)', () => {
       ]);
     });
   });
-  it('cleans up DOM', () => {
+  it('cleans up DOM', async () => {
     getMultipleTextDimensions({
       texts: [SAMPLE_TEXT[0], SAMPLE_TEXT[1]],
     });
 
     expect(document.querySelectorAll('svg')).toHaveLength(1);
-
-    return promiseTimeout(() => {
-      expect(document.querySelector('svg')).toBeNull();
-    }, 600);
+    await promiseTimeout(() => {}, 501);
```

Comment:
this is a case where it seems like we would want to use fake timers, otherwise it adds `0.5s` to the tests? I'm sure there were other edge cases.

Reply:
I tried but couldn't get this one to work with fake timers. The underlying logic that uses `setTimeout` has a debouncing effect + dependency tracking counter that is a bit complicated. Prefer to leave it like this for now.

Reply:
I open this as a separate issue and will resolve this PR for now. https://github.com/apache-superset/superset-ui/issues/286